File: leetcode/cpp/728_Self_Dividing_Numbers.cpp
#include <vector>
using namespace std;
class Solution {
public:
	vector<int> selfDividingNumbers(int left, int right) {
		vector<int> ret;
		for (int i = left; i <= right; ++i) {
			string str = to_string(i);
			bool flag = true;
			for (char c : str) {
				if (c == '0') {
					flag = false;
					break;
				}

				if (i % (c - '0') != 0) {
					flag = false;
					break;
				}
			}

			if (flag) ret.push_back(i);
		}
		return ret;
	}
};

File: leetcode/cpp/728_Self_Dividing_Numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        ans = []

        for num in range(left, right + 1):
            if all(d != '0' and num % int(d) == 0 for d in str(num)):
                ans.append(num)

        return ans  
